src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs

```diff
@@ -505,6 +506,40 @@ public async Task PersistentState_StreamsReused_StatePersisted()
             Assert.Equal(true, state);
         }
 
+        [ConditionalFact]
+        [MsQuicSupported]
+        public async Task CertificatePassedToQuicConnectionContext()
+        {
+            // Arrange
+            await using var connectionListener = await QuicTestHelpers.CreateConnectionListenerFactory(LoggerFactory, clientCertificateRequired: true);
+
+            var options = QuicTestHelpers.CreateClientConnectionOptions(connectionListener.EndPoint);
+            var testCert = TestResources.GetTestCertificate();
+            options.ClientAuthenticationOptions.ClientCertificates = new X509CertificateCollection { testCert };
+            using var quicConnection = new QuicConnection(QuicImplementationProviders.MsQuic, options);
+            await quicConnection.ConnectAsync().DefaultTimeout();
+
+            var serverConnection = await connectionListener.AcceptAndAddFeatureAsync().DefaultTimeout();
+
+            /*var tlsFeature = serverConnection.Features.Get<ITlsConnectionFeature>();
```

Comment:
@JamesNK the `ITlsConnectionFeature` and `IProtocolErrorCodeFeature` are not available on the `serverConnection`, only the `serverStream`. These tests also only try to get the `IProtocolErrorCodeFeature` from the stream - is that expected?

https://github.com/dotnet/aspnetcore/blob/709038b45159227a422807f26d6d41dd7707ea10/src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs#L335

Reply:
ITlsConnectionFeature and IProtocolErrorCodeFeature should be available on the connection.

https://github.com/dotnet/aspnetcore/blob/45222c12d944a8915ba480220de8a1f9c133c0c4/src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionContext.FeatureCollection.cs#L29-L33

Reply:
If the feature is not available on the connection, why does this test code work?

https://github.com/dotnet/aspnetcore/blob/709038b45159227a422807f26d6d41dd7707ea10/src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs#L335

Reply:
That's getting the feature from the stream - the code above at line 524 does not have either `ITlsConnectionFeature` or `IProtocolErrorCodeFeature` available in the featureCollection of the connection.

Reply:
> Have you debugged into what happens when Get is called?

This is weird - under the debugger, I get the feature and it has the ClientCertificate set, and the test passes. But when run normally, the test fails at line 526 at `Assert.NotNull(tlsFeature.ClientCertificate)`...